units/patch_intent.go

```diff
@@ -485,6 +488,37 @@ func (j *patchIntentProcessor) createGitHubSubscriptions(p *patch.Patch) error {
 	return catcher.Resolve()
 }
 
+// createGithubMergeSubscription creates a subscription on a commit for the GitHub merge queue.
+func (j *patchIntentProcessor) createGitHubMergeSubscription(p *patch.Patch) error {
+	catcher := grip.NewBasicCatcher()
+	ghSub := event.NewGithubCheckAPISubscriber(event.GithubCheckSubscriber{
+		Owner: p.GithubMergeData.Org,
+		Repo:  p.GithubMergeData.Repo,
+		Ref:   p.GithubMergeData.HeadSHA,
+	})
+
+	patchSub := event.NewExpiringPatchOutcomeSubscription(j.PatchID.Hex(), ghSub)
+	if err := patchSub.Upsert(); err != nil {
+		catcher.Wrap(err, "inserting patch subscription for GitHub merge queue")
+	}
+	buildSub := event.NewExpiringBuildOutcomeSubscriptionByVersion(j.PatchID.Hex(), ghSub)
+	catcher.Wrap(buildSub.Upsert(), "inserting build subscription for GitHub merge queue")
+	input := thirdparty.SendGithubStatusInput{
+		VersionId: j.PatchID.Hex(),
+		Owner:     p.GithubMergeData.Org,
+		Repo:      p.GithubMergeData.Repo,
+		Ref:       p.GithubMergeData.HeadSHA,
+		Desc:      "patch created",
+		Caller:    j.Name,
+		Context:   "evergreen",
+	}
+	err := thirdparty.SendPendingStatusToGithub(input, j.env.Settings().Ui.Url)
```

Comment:
Can you please explain the `j.env.Settings().Ui.Url` a bit? What do we expect it to be? Why do we want it over uiConfig.Get(env).Url?

Reply:
They're the same value. `uiConfig.Get()` makes a database call. Since we already have the environment object, we don't need to make the call.